### ISSUE TITLE
No connection with SMA

### ISSUE THREAD

**Issue by wannestal** [kurtwerbrouck/SMA-SunnyBoy-Python#1]
Hi, just tried your code in visual studio code.
I get:
HTTPConnectionPool(host='192.168.0.180', port=80): Max retries exceeded with url: /dyn/login.json (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x000001AA18253580>: Failed to establish a new connection: [WinError 10060]

I am sure that settings are ok: ip adress, password. For user, I filled in "user".
I have a Sunny Boy 3000TL-21 405

Kind regards
Wannes

**Comment by kurtwerbrouck** [comment]
Hello

Try as user : usr

Kurt

Verstuurd vanaf mijn iPhone

Op 22 feb. 2022 om 10:48 heeft wannestal ***@***.***> het volgende geschreven:

﻿

Hi, just tried your code in visual studio code.
I get:
HTTPConnectionPool(host='192.168.0.180', port=80): Max retries exceeded with url: /dyn/login.json (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x000001AA18253580>: Failed to establish a new connection: [WinError 10060]

I am sure that settings are ok: ip adress, password. For user, I filled in "user".
I have a Sunny Boy 3000TL-21 405

Kind regards
Wannes

**Comment by kurtwerbrouck** [comment]
payload = ('{"pass" : "password", "right" : "usr"}')
________________________________
Van: wannestal ***@***.***>
Verzonden: dinsdag 22 februari 2022 10:48
Aan: kurtwerbrouck/SMA-SunnyBoy-Python ***@***.***>
CC: Subscribed ***@***.***>
Onderwerp: [kurtwerbrouck/SMA-SunnyBoy-Python] No connection with SMA (Issue #1)


Hi, just tried your code in visual studio code.
I get:
HTTPConnectionPool(host='192.168.0.180', port=80): Max retries exceeded with url: /dyn/login.json (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x000001AA18253580>: Failed to establish a new connection: [WinError 10060]

I am sure that settings are ok: ip adress, password. For user, I filled in "user".
I have a Sunny Boy 3000TL-21 405

Kind regards
Wannes

**Comment by wannestal** [comment]
No, the issue persists. When I go from a internet browser to the ip address of my Sunny Boy, it finds nothing; should I get something? Sunny explorer and Sunny portal: no problem to connect.

**Comment by kurtwerbrouck** [comment]
hello

Keep in mind that it can take up to 45 sec before the sma send an SID reply to the program- also , i have noticed that when i am logged in to the webpage that i cant get a connection via python.

i send this evening a video of the python program that is reading the data

I also have it in Micropython and Arduino and in ESP32


________________________________
Van: wannestal ***@***.***>
Verzonden: dinsdag 22 februari 2022 12:29
Aan: kurtwerbrouck/SMA-SunnyBoy-Python ***@***.***>
CC: kurtwerbrouck ***@***.***>; Comment ***@***.***>
Onderwerp: Re: [kurtwerbrouck/SMA-SunnyBoy-Python] No connection with SMA (Issue #1)


No, the issue persists. When I go from a internet browser to the ip address of my Sunny Boy, it finds nothing; should I get something? Sunny explorer and Sunny portal: no problem to connect.

**Comment by wannestal** [comment]
Ok, thank you!
In fact: to get it working on an esp32 would be the best solution for me.

**Comment by kurtwerbrouck** [comment]
hello

arduino version - values are displayed on a OLED display


________________________________
Van: wannestal ***@***.***>
Verzonden: dinsdag 22 februari 2022 12:39
Aan: kurtwerbrouck/SMA-SunnyBoy-Python ***@***.***>
CC: kurtwerbrouck ***@***.***>; Comment ***@***.***>
Onderwerp: Re: [kurtwerbrouck/SMA-SunnyBoy-Python] No connection with SMA (Issue #1)


Ok, thank you!
In fact: to get it working on an esp32 would be the best solution for me.

**Comment by kurtwerbrouck** [comment]
reply from the python script

[cid:8b1c12df-e96e-4084-bf2f-3ccb3b4d25e5]


________________________________
Van: wannestal ***@***.***>
Verzonden: dinsdag 22 februari 2022 12:39
Aan: kurtwerbrouck/SMA-SunnyBoy-Python ***@***.***>
CC: kurtwerbrouck ***@***.***>; Comment ***@***.***>
Onderwerp: Re: [kurtwerbrouck/SMA-SunnyBoy-Python] No connection with SMA (Issue #1)


Ok, thank you!
In fact: to get it working on an esp32 would be the best solution for me.

**Comment by wannestal** [comment]
I have to investigate this further...
Maybe modbus is another possibility to get the data that I want.

**Comment by wannestal** [comment]
I am trying your code on a ESP32 with micropython. Have installed the required packages, but now I am stuck on 'ImportError: can't import name NullHandler' while debugging.

**Comment by kurtwerbrouck** [comment]
hello

see example

it took me also some weeks to get this up and running on python - micropython and c++ on the esp32

Kurt
________________________________
Van: wannestal ***@***.***>
Verzonden: woensdag 23 februari 2022 20:51
Aan: kurtwerbrouck/SMA-SunnyBoy-Python ***@***.***>
CC: kurtwerbrouck ***@***.***>; Comment ***@***.***>
Onderwerp: Re: [kurtwerbrouck/SMA-SunnyBoy-Python] No connection with SMA (Issue #1)


I am trying your code on a ESP32 with micropython. Have installed the required packages, but now I am stuck on 'ImportError: can't import name NullHandler' while debugging.
